src/libs/SelectionScraper/index.ts

```diff
@@ -96,40 +100,41 @@ const getHTMLOfSelection = () => {
 
 /**
  * Clears all attributes from dom elements
- * @param {Object} dom htmlparser2 dom representation
- * @param {Boolean} isChildOfEditorElement
- * @returns {Object} htmlparser2 dom representation
+ * @param dom - dom htmlparser2 dom representation
  */
-const replaceNodes = (dom, isChildOfEditorElement) => {
-    let domName = dom.name;
-    let domChildren;
-    const domAttribs = {};
-    let data;
+const replaceNodes = (dom: Node, isChildOfEditorElement: boolean): Node => {
+    let domName;
+    let domChildren: Node[] = [];
+    const domAttribs: Element['attribs'] = {};
+    let data = '';
 
     // Encoding HTML chars '< >' in the text, because any HTML will be removed in stripHTML method.
-    if (dom.type === 'text') {
+    if (dom.type.toString() === 'text' && dom instanceof DataNode) {
         data = Str.htmlEncode(dom.data);
     }
 
-    // We are skipping elements which has html and body in data-testid, since ExpensiMark can't parse it. Also this data
-    // has no meaning for us.
-    if (dom.attribs && dom.attribs[tagAttribute]) {
-        if (!elementsWillBeSkipped.includes(dom.attribs[tagAttribute])) {
-            domName = dom.attribs[tagAttribute];
+    if (dom instanceof Element) {
```

Comment:
This condition is added to [narrow down the type](https://www.typescriptlang.org/docs/handbook/2/narrowing.html) so we can use some of the properties in the if body.

> This will allow the function to return garbage value

Not exactly sure what you mean by this? We tested the logic and the argument is either `DataNode` or `Element`, so depending on the argument type the code from first/second if is executed

Reply:
We need to narrow the type, but you point is also valid. I'm proposing something like this: 
```ts
    if (dom.type === ElementType.Text && dom instanceof DataNode) {
        ...
    } else if (dom instanceof Element) {
        ...
    } else {
        throw new Error(`Unknown dom type: ${dom.type}`);
    }
```

We would have to test it thoroughly, but it's the best what can be done. Any other ideas? @parasharrajat @Skalakid

Reply:
I agree with that